xwiki-platform-core/xwiki-platform-sharepage/xwiki-platform-sharepage-ui/src/main/resources/XWiki/SharePage.xml

```diff
@@ -337,7 +337,7 @@ $!{escapetool.xml($message)}
 &lt;/pre&gt;
 #end
 
-&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${doc.getExternalURL()}&lt;/a&gt; in your browser.&lt;/p&gt;
+&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${util.decodeURI("${doc.getExternalURL()}")}&lt;/a&gt; in your browser.&lt;/p&gt;
```

Comment:
I don't understand why it shouldn't be encoded. It's an URL, it should be encoded, otherwise copy/pasting it wouldn't work all the time.

The only case where it would have to be decoded is it is going to be encoded a second time, but that's not the case. So, what's the problem that you're trying to solve?